test/testdata/lsp/fast_path/static_field_change_type__use.1.rbupdate

```diff
@@ -0,0 +1,8 @@
+# typed: strict
+# assert-fast-path: static_field_change_type__def.rb,static_field_change_type__use.rb
+extend T::Sig
+
+sig {returns(String)}
+def foo
+  STATIC_FIELD.fetch(:key, "default") + "suffix"
```

Comment:
non-blocking, but it's worth noting that these `.fetch` calls could start to succeed if somehow we introduced a bug making it seem like `STATIC_FIELD` was `T.untyped` on fast path updates.

Might be worth adding a `T.reveal_type` in these files just to double check?